### DISCUSSION TITLE
Is there priority between cname and default blocking?

### DISCUSSION THREAD

**Discussion by uNitrous** [AdguardTeam/AdGuardHome#2770] [Q&A] [answered]
### Problem Description
eg: issues.net (domain) and it has cname discussion.org

A list has a rule to block discussion.org
B list has a rule to block issues.net
consider AGH has some other blocklists, custom filtering rules.

Which list rule AGH take first to block the issues.net domain? If it B, is there any chance for affect to it?

**Comment by uNitrous** [comment]
i ask because when see any site corrupted next i dont want to mistake and unblock trackers, ads.

**Comment by uNitrous** [comment]
2. why this can't block? from https://github.com/AdguardTeam/AdGuardHome/discussions/2759#discussioncomment-408153 as i see AGH cant block it.
https://1.1.1.1/

**Reply by ameshkov** [reply]
The thing is there is no DNS query when you visit http://1.1.1.1 -- because there's no domain, you're visiting the specific IP.

**Comment by ameshkov** [comment, marked as answer]
Default blocking will always have higher priority.

The difference is that those rules are checked without actually making the DNS query, and for blocking by CNAME/IP you need to have a DNS response as well.
